src/Raven.Client/Documents/Operations/CollectionStatistics.cs

```diff
@@ -1,17 +1,70 @@
 ﻿using System.Collections.Generic;
+using Raven.Client.Util;
+using Sparrow.Json.Parsing;
 
 namespace Raven.Client.Documents.Operations
 {
-    public class CollectionStatistics
+    public class CollectionStatistics : IDynamicJson
     {
         public CollectionStatistics()
         {
             Collections = new Dictionary<string, long>();
         }
 
-        public int CountOfDocuments { get; set; }
-        public int CountOfConflicts { get; set; }
+        public long CountOfDocuments { get; set; }
+        public long CountOfConflicts { get; set; }
 
         public Dictionary<string, long> Collections { get; set; }
+
+        public virtual DynamicJsonValue ToJson()
+        {
+            return new DynamicJsonValue
+            {
+                [nameof(CountOfDocuments)] = CountOfDocuments,
+                [nameof(CountOfConflicts)] = CountOfConflicts,
+                [nameof(Collections)] = DynamicJsonValue.Convert(Collections)
+            };
+        }
+    }
+
+    public class DetailedCollectionStatistics : CollectionStatistics
+    {
+        public DetailedCollectionStatistics()
+        {
+            ExtendedCollectionDetails = new Dictionary<string, CollectionDetails>();
+        }
+
+        public Dictionary<string, CollectionDetails> ExtendedCollectionDetails { get; set; }
```

Comment:
In this case, since we already moved that to other class and endpoint, maybe lets not inherit and substitute 'Collections' from 'CollectionStatistics' with `Dictionary<string, CollectionDetails>` and put there the etag as well.

Reply:
Sorry, I didn't quite understand. I will remove the inheritance, but what do you mean by "substitute 'Collections' from 'CollectionStatistics' with Dictionary<string, CollectionDetails> and put there the etag as well"?

Did you mean I should remove the 'Collection' property from the 'DetailedCollectionStatistics' class, because the 'CollectionDetails' class also contains the number of documents?

Reply:
If we drop the inheritance between these classes, then we don't need `ExtendedCollectionDetails`, we can call the property `Collections`. That would make things simpler.